scripts/pin_the_slow_way.py

```diff
@@ -92,7 +92,7 @@
           'tk': 'tk 8.5.*',  # 8.5.18
           'vlfeat': 'vlfeat 0.9.20',  # NA
           'xz': 'xz 5.2.*',  # 5.2.2
-          'zlib': 'zlib 1.2.8',  # 1.2.8
+          'zlib': 'zlib >=1.2.11,<1.3.0a0',  # zlib >=1.2.11,<1.3.0a0
```

Comment:
This is correct only until `1.2.12` comes out.

Reply:
Do you want to be more conservative and do `zlib 1.2.11` only?
I guess we cannot predict what is going to happen, but if `1.3.0a0` is OK why wouldn't `1.2.12`  be OK too :grimacing: 

PS: I don't know where the info for `1.3.0a0` comes from. It is not in https://abi-laboratory.pro/tracker/timeline/zlib/

Reply:
It's about build vs run-time pinning. When 1.2.12 comes out, if a package is compiled with 1.2.11, it will be pinned as `>=1.2.11,<1.3` and if it is compiled with 1.2.12, it will be pinned as `>=1.2.12,<1.3`. So `run_exports` does the correct thing. Ours is not because we have the same pinning for build and run

Reply:
That I know, what we need is a decision to assume that a build and run `>=1.2.11,<1.3.0a0` is an OK risk or if we should be conservative and do `1.2.11`.

Manually doing build/run different is too time consuming now. I don't think it is worth it.
Our efforts should be on moving to conda-build 3 and use `run_exports`.

Reply:
@isuruf you're right.  The danger is that if someone built when 1.2.12 was available, and then someone installed that package, if they already had 1.2.11 installed, conda would not necessarily update it, and then the new package built against 1.2.12 may not work.

@ocefpaf 1.3.0a0 comes from incrementing 1.2, and adding a0 because conda considers prerelease versions to be less than the actual release number.  <1.3 looks nice, but can bring in 1.3.0rc0, 1.3.0beta, etc.  I've been bitten by that, and wow is it confusing.  That's why cb3 is adding that stuff.

Reply:
OK. So let's be conservative and not risk that.